### PR TITLE
infra/`generate_skels.py`: Support Python-style comments

### PR DESCRIPTION
**Pull request by NickZaharia308**
Modifies copies of `generate_skels.py` to support `.py` files.

# Prerequisite Checklist

- [ ] Read the [contribution guidelines](https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [ ] Tested your changes against relevant architectures and platforms;
- [ ] Updated relevant documentation (if needed).

### CODE REVIEW

File: chapters/data/process-memory/drills/tasks/copy/generate_skels.py [hunk 1]
```diff
@@ -100,43 +99,49 @@ def main():
     args = parser.parse_args()
 
     for root, dirs, files in os.walk(args.input):
-        new_root = os.path.join(args.output, os.path.relpath(root, args.input))
+        new_root = os.path.join(args.output, root.replace(args.input, ""))
```
Comment (Alex-deVis) [inline chapters/data/process-memory/drills/tasks/copy/generate_skels.py:102]: Is this better or recommended? If you found corner cases for the original approach, please refer them.

**Review by Alex-deVis — CHANGES_REQUESTED** [review]
Thank you for this! Before we can merge it, you'll have to solve the inline comments.
You removed the license tag and the `r` prefix from the regexes. These changes were added recently so I assume you have not rebased.

File: chapters/data/process-memory/drills/tasks/copy/generate_skels.py [hunk 2]
```diff
@@ -148,3 +149,4 @@ def main():
 
 if __name__ == "__main__":
     sys.exit(main())
+
```
Comment (Alex-deVis) [inline chapters/data/process-memory/drills/tasks/copy/generate_skels.py:152]: The file ends with a new line already. This is why superlinter complains.
```suggestion
```

**Review by Alex-deVis — CHANGES_REQUESTED** [review]
Remove the extra newlines and we can merge this.

File: chapters/data/process-memory/drills/tasks/copy/generate_skels.py (hunk 2 above)
Comment (NickZaharia308) [reply chapters/data/process-memory/drills/tasks/copy/generate_skels.py:152]: Great, thanks!

**Comment by NickZaharia308** [comment]
Should I make these changes to the private repo as well?

**Comment by teodutu** [comment]
> Should I make these changes to the private repo as well?

Yes please.

**Review by teodutu — APPROVED** [review]
Looks fine to me.

**Review by Alex-deVis — APPROVED** [review]

**Pull request merged**